packages/expo-modules-core/android/build.gradle

```diff
@@ -84,6 +85,24 @@ if (findProject(":app")) {
   FOR_HERMES = (findProperty('expo.jsEngine') ?: "jsc") == "hermes"
 }
 
+if (FOR_HERMES) {
+  // The `hermes-engine` package doesn't contain the correct version of the Hermes.
+  // The AAR we need to import is located in the React Native package.
+  // However, the version from RN doesn't include header files.
+  // To get those, we must fall back to the `hermes-engine` package.
+  HERMES_ENGINE_DIR = new File(["node", "--print", "require.resolve('hermes-engine/package.json')"].execute(null, rootDir).text.trim()).parent
+  if (REACT_NATIVE_BUILD_FROM_SOURCE) {
+    HERMES_AAR = file("$HERMES_ENGINE_DIR/android/hermes-debug.aar")
```

Comment:
for building from source, maybe we should have a way to extract the `:ReactAndroid:hermes-engine` libhermes.so. this is not quite correct to get from npm.
it doesn't matter in the mean time. maybe leaving a TODO comment here is enough.